docs/advanced_usage/reporting.rst

```diff
@@ -0,0 +1,117 @@
+
+Pykiso Reporting
+================
+
+.. _junit_report:
+
+JUnit XML report
+~~~~~~~~~~~~~~~~
+
+Generating a JUnit XML report
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For CI integration, JUnit reports in XML format can be generated at the end of the test session.
+Simply invoke ``pykiso`` with:
+
+.. code:: bash
+
+    pykiso -c my_config.yaml --junit=./path/to/my_report.xml
+
+The resulting report will contain:
+
+- One ``testsuite`` for each ``pykiso.BasicTest``, ``pykiso.BasicTestSuiteSetup``, ``pykiso.BasicTestSuiteTeardown``.
+- One ``testcase`` inside the ``testsuite`` for each test method. Using `subTest <https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests>`_
+  will also append a ``testcase`` with modified test case name ``"test_method [my subtest description]"``.
+
+.. note:: The captured logs will always appear in the ``testcase`` section.
+
+Adding contextual information to the JUnit report
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Any user-defined information can be added to all test cases of a test suite as JUnit ``properties``.
```

Comment:
Is it a self idea or does it come from junit reference manual?

Reply:
This comes from the Junit reference :)